### ISSUE TITLE
Improve default lighting.

### ISSUE THREAD

**Issue by donmccurdy** [donmccurdy/three-gltf-viewer#44]
I'm using three-point lighting now, but it doesn't seem to generalize especially well, even to the https://github.com/KhronosGroup/glTF-Sample-Models/ examples. Would appreciate suggestions/PRs to improve this.

**Comment by zellski** [comment]
I've battled with this a fair bit in our three.js-based model previewer. We now use roughly what you have, and honestly I'm often unhappy with it myself. I'd like generally more fill light.

For PBR materials, a HDR environment map seems to make a big difference. I've also been curious why BabylonJS tends to end up with generally brighter views (perhaps at the cost of contrast).

Have you identified a common thread between models that you feel are insufficiently lit?

**Comment by donmccurdy** [comment]
Yeah the environment map is pretty much required for PBR to look right.

Also there was also a bug until recently (bug is in <=r87, fixed in `dev` branch of three.js) where baseColor and emissive textures were applied with linear encoding, when they should have been sRGB. The bug made models look generally brighter and more washed out.

I think across the board everything feels under-lit, but I'd been hoping to find a proper lighting guide before spending time tweaking things arbitrarily.

**Comment by zellski** [comment]
Wow, thanks for pointing out that encoding bug. That may make a significant difference for us. Any guesses as to when r88 may be out? :-)

I guess where I was going with the environment map comment is... and apologies if I am stating the painfully obvious... I'm increasingly thinking that what I want -- for the model to be well-lit at all times, from all directions, without resorting to overly much simple ambient -- simply can't be done adequately, not with the simple lights system we have at our disposal.

Neither directional nor spots really do what we would expect lights to do in real life (except perhaps in outer space). Real light bounces endlessly in the environment. We don't have global illumination, but environment maps -- HDR specifically -- can provide a simulation thereof.

Perhaps the majority of light energy in the scene should come from there, and directional exists only to provide additional accent and structure?

**Comment by donmccurdy** [comment]
Here's a particular example: https://sketchfab.com/models/f2f13a8630004b1c82730d8b9ffa0e1f#

Looks great on sketchfab but in my viewer the bear is too dark, and the balloons too bright, and no tweaking of knobs seems to really fix that. 😕

**Comment by zellski** [comment]
I feel if we had more knobs, we might be able to dig deeper into this mystery. Per-material, per-mesh, per-light dials. Tweaking roughness/metallic values in the .gltf download from Sketchfab, and turning the lights way up, that seems to help some. That does admittedly feel somewhat artificial.

Dropped you an email with couple of screenshots for further discussion.

**Comment by donmccurdy** [comment]
Some suggestions in this thread: https://github.com/mrdoob/three.js/issues/12554#issuecomment-341686730

>  I did some digging and think I gained some surface-level knowledge on this topic of color management. I think three.js utilizes linear color space throughout it's pipeline for lighting calculations, etc. So when you flag the map/texture as sRGB, it removes the gamma correction so calculations can correctly be done in linear color space. I think this is the right approach and is followed by most standard 3D applications. However, by default, three.js doesn't gamma correct it's rendering output by default. So removing the gamma correction in sRGB texture results in a darker than intended output.

```js
renderer.gammaFactor = 2.2;
renderer.gammaOutput = true;
```

**Comment by DJviolin** [comment]
Any update on this? The `Mixed Reality Viewer` in Win10 also can give a pleasant viewing experience. In my setup the 2.2 gamma is an overkill or I didn't figured out how to utilitize the ambient, lights.

**Comment by donmccurdy** [comment]
Updates:

First, I can't see any difference whatsoever when changing the gamma settings... anyone having better luck with that? Does that need to happen before rendering starts or something?

Second, let's try adding more knobs. I've updated the viewer with some new controls:

1. Texture encoding for `baseColor` and `emissive` properties. Default is sRGB.
2. Exposure.
3. Simplified to 1 ambient + 1 direct light, with direct control of intensity and color for each.

My immediate takeaway is that everything looks (much) better lit when texture encoding is set to Linear. It's [_not supposed_ to be linear](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#pbrmetallicroughnessbasecolortexture), though. And it does look more washed out that way. But with the "Bear and balloon" example from Sketchfab ([above](https://github.com/donmccurdy/three-gltf-viewer/issues/44#issuecomment-340004116)), to get reasonable-looking results in sRGB I have to set `directIntensity=2` `ambientIntensity=1`, and that's higher than I expect for three.js, and higher than the [qtek viewer uses](https://github.com/pissang/qtek-model-viewer/blob/8df900b02306589258eb562bb4738ee1d78e1455/editor/src/getDefaultSceneConfig.js#L30-L97) (while getting better results).

With all default settings (`directIntensity=0.8`, `ambientIntensity=0.3`), here is a comparison of texture encodings:

| sRGB | Linear |
|---|---|
| ![srgb](https://user-images.githubusercontent.com/1848368/33236417-a31b8d32-d207-11e7-9dec-3f3395c9a546.png) | ![linear](https://user-images.githubusercontent.com/1848368/33236418-a4b18e12-d207-11e7-935f-6e50fe48de2d.png) |



We could try adding an HDR envMap, I suppose, but it feels like that shouldn't be the underlying issue here... and if "more light" really is the answer, is mixing Phong/Lambert materials in with glTF PBR materials just out of the question then?

**Comment by donmccurdy** [comment]
Added gamma settings to the dropdown temporarily for good measure...

**Comment by donmccurdy** [comment]
More updates: [I need to learn how gamma correction works...](https://github.com/mrdoob/three.js/issues/11110#issuecomment-346983451) after setting `gammaOutput=true` correctly this time, things look much better.

**Comment by donmccurdy** [comment]
Going to consider this closed, suggestions for more general improvements still welcome.

**Comment by Rayfoundry** [comment]
Setting renderer.gammaOutput = true; works with the GLTF loader if you are using textures. But I haven't found a solution yet for models only using just vertex colors. Setting the flag makes those models display much brighter. Any suggestions? #1110

**Comment by donmccurdy** [comment]
@Rayfoundry could you open a new issue and provide a bit more detail — where are your vertex models coming from (Blender, converted from another format, etc.), and what are they "much brighter" compared to?  Thanks!
